packages/mapper/src/mapper.ts

```diff
@@ -1580,4 +1580,37 @@ export default class Mapper {
     const stats = await this.stats(collection);
     return (stats.storageSize || 0) + (stats.totalIndexSize || 0);
   }
+
+  /**
+   * Drop a collection.
+   *
+   * @param {Collection} collection - The collection.
+   * @return {Promise} returns Promise
+   */
+  async drop(
+    collection: Collection,
+  ): Promise<boolean> {
+    this.messageBus.emit(
+      'mongosh:api-call',
+      {
+        method: 'drop',
+        class: 'Collection',
+        db: collection._database._name,
+        coll: collection._name
+      }
+    );
+
+    try {
+      return await this.serviceProvider.dropCollection(
+        collection._database._name,
+        collection._name
+      );
+    } catch (error) {
+      if (error.codeName === 'NamespaceNotFound') {
+        return false;
```

Comment:
we should emit an error for the logger here. Might be useful for people debugging:
```js
this.messageBus.emit('mongosh:error', new Error(`Namespace not found: ${collection._name}`));
```

I am also wondering if we should maybe also return something to user indicating that a collection could not be dropped.